examples/light-switch-app/silabs/src/LightSwitchMgr.cpp

```diff
@@ -26,18 +26,131 @@
 
 #include "AppConfig.h"
 #include "AppEvent.h"
+#include "AppTask.h"
 
 #include <app-common/zap-generated/attributes/Accessors.h>
 #include <app-common/zap-generated/cluster-objects.h>
 #include <app/clusters/switch-server/switch-server.h>
 #include <platform/CHIPDeviceLayer.h>
+#include <cmsis_os2.h>
+#include <platform/silabs/platformAbstraction/SilabsPlatform.h>
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer;
+using namespace chip::DeviceLayer::Silabs;
+
+#define LONG_PRESS_TIMEOUT 3000
+#define STEP_SIZE 1
+#define TRANSITION_TIME 0
+#define OPTIONS_MASK 0
+#define OPTIONS_OVERRIDE 0
 
 LightSwitchMgr LightSwitchMgr::sSwitch;
 
+LightSwitchMgr::LightSwitchMgr() {}
+
+AppEvent CreateNewEvent(AppEvent::AppEventTypes type)
+{
+    AppEvent aEvent;
+    aEvent.Type                      = type;
+    aEvent.Handler                   = LightSwitchMgr::GeneralEventHandler;
+    LightSwitchMgr * lightSwitch         = static_cast<LightSwitchMgr *>(&LightSwitchMgr::GetInstance());
```

Comment:
This applies to other locations as well
```suggestion
    LightSwitchMgr * lightSwitch         = &LightSwitchMgr::GetInstance();
```

Reply:
modified it in this file